src/components/TImeLeftBar/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react'
+import { Progress } from 'antd'
+import PropTypes from 'prop-types'
+
+export const TimeLeftBar = ({
+  duration,
+  colorone,
+  colortwo,
+  coloronethreshold,
+  className,
+  ...props
+}) => {
+  const [progressPercent, setProgressPercent] = useState(0)
+  useEffect(() => {
+    let startTime = new Date().getTime()
+
+    const interval = setInterval(() => {
+      const currentTime = new Date().getTime()
+      const elapsedTime = currentTime - startTime
+
+      if (elapsedTime >= duration) {
+        clearInterval(interval)
+      }
+      // calculating the percentage of time left
+      const percentage = 100 - (elapsedTime / duration) * 100
+      setProgressPercent(percentage)
```

Comment:
Also if percentage is < 0, we should stop the timer.

Reply:
Added a condition for it